packages/arb-token-bridge-ui/src/hooks/useNetworksAndSigners.tsx

```diff
@@ -142,45 +141,33 @@ export function NetworksAndSignersProvider(
   // TODO: Don't run all of this when an account switch happens. Just derive signers from networks?
   const update = useCallback(
     async (web3Provider: Web3Provider, address: string) => {
-      // It's safe to fall back to 0, as invalid values will be ignored by the SDK
-      updateL2ChainIdAndClearCache(selectedL2ChainId ?? 0)
 
-      async function isSwitchingToPartnerNetwork() {
-        const nextChainId = (await web3Provider.getNetwork()).chainId
-        const current = latestResult.current
-
-        if (current.status !== UseNetworksAndSignersStatus.CONNECTED) {
-          return false
-        }
-
-        if (current.isConnectedToArbitrum) {
-          return nextChainId === current.l2.network.partnerChainID
+      async function defaultL2NetworkChainId(provider: JsonRpcProvider) {
+        const providerChainId = (await web3Provider.getNetwork()).chainId
+        switch(providerChainId){
+          case ChainId.Mainnet:
+          case ChainId.ArbitrumOne:
+            return ChainId.ArbitrumOne
+          case ChainId.Rinkeby:
+          case ChainId.ArbitrumRinkeby:
+            return ChainId.ArbitrumRinkeby
+          case ChainId.Goerli:
+          case ChainId.ArbitrumGoerli:
+            return ChainId.ArbitrumGoerli
+          case ChainId.ArbitrumNova:
+            return ChainId.ArbitrumNova
+          default:
+            return providerChainId
```

Comment:
```suggestion
          default:
            const exhaustiveCheck: never = providerChainId;
            throw new Error(`Unhandled chain id: ${exhaustiveCheck}`);
```

Reply:
this would break in something like the UI doing a `addCustomNetwork`
It would be nice to move this logic to `packages/arb-token-bridge-ui/src/util/networks.ts`